tools/tune/src/src_generate.m

```diff
@@ -274,3 +388,80 @@
         cstr = sprintf('%s%s', cstr1, cstr2);
 end
 end
+
+function delta = test_proto_src(src1, src2, cfg)
+
+if src1.filter_length < 1
+	delta = 0;
+	return;
+end
+
+[fs1, fs2] = src_fs12(src1, src2);
+t = 1.0;
+t_ignore = 0.5;
+a = 10^(-1/20);
+f = [211 1234 3500 min(20/48 * min(fs1, fs2), 20e3)];
+for i = 1:length(f)
+	thdn(i) = src_thdn(src1, src2, f(i), a, t_ignore, t);
+end
+
+delta =  max(thdn) - cfg.thdn;
+
+end
+
+function [fs1, fs2] = src_fs12(src1, src2)
+
+fs1 = src1.fs1;
+if src2.filter_length > 0
+	fs2 = src2.fs2;
+else
+	fs2 = src1.fs2;
+end
+
+end
+
+function thdn = src_thdn(src1, src2, f, a, t_ignore, t)
+
+[fs1, fs2] = src_fs12(src1, src2);
+x = multitone(fs1, f, a, t);
+y2 = proto_src(src1, src2, x);
+
+% Apply standard low-pass for higher output rates
+if fs2 > 42e3
+	y2 = stdlpf(y2, 20e3, fs2);
+end
+
+y3 = stdnotch(y2, f, fs2);
+idx = round(t_ignore * fs2);
+level_total = 10^(a / 20);
```

Comment:
Mistake, should be 20*log10(a)

Reply:
This is fixed now. Also the change includes THD+N check with octave spaced log sweep.